packages/gatsby/src/utils/page-data.ts

```diff
@@ -119,36 +114,21 @@ export async function readPageQueryResult(
 
 export async function writePageData(
   publicDir: string,
-  {
-    componentChunkName,
-    matchPath,
-    path: pagePath,
-    staticQueryHashes,
-  }: IPageData
+  pageData: IPageData
 ): Promise<string> {
-  const result = await readPageQueryResult(publicDir, pagePath)
-
-  const outputFilePath = generatePageDataPath(publicDir, pagePath)
-  let body = `{
-    "componentChunkName": "${componentChunkName}",
-    "path": "${pagePath}",
```

Comment:
_offtopic for this PR_:  quotes are not escaped here. I wonder if it is the cause of https://github.com/gatsbyjs/gatsby/issues/32892

Reply:
I think with https://github.com/gatsbyjs/gatsby/pull/33067/commits/90a2c57faa70e7fe5420d44237e6245b68cc1985 we can resolve this one?

Reply:
Yup 👍